Userland/Libraries/LibGfx/AntiAliasingPainter.cpp

```diff
@@ -169,3 +170,218 @@ void Gfx::AntiAliasingPainter::draw_cubic_bezier_curve(const FloatPoint& control
         draw_line(fp1, fp2, color, thickness, style);
     });
 }
+
+void Gfx::AntiAliasingPainter::draw_circle(IntPoint center, int radius, Color color)
+{
+    /*
+    Algorithm from: https://cs.uwaterloo.ca/research/tr/1984/CS-84-38.pdf
+    Inline comments are from the paper.
+    */
+
+    // TODO: Generalize to ellipses (see paper)
+
+    constexpr int subpixel_resoulution = 255;
+    const int intensity { color.alpha() };
+
+    // Note: Variable names below are from the paper
+
+    /* Current pixel address */
+    int i { 0 }, q { radius };
+
+    /* 1st and 2nd order differences of y*/
+    int delta_y { 0 }, delta2_y { 0 };
+
+    /* Exact and predicted values of f(i) */
+    int y { subpixel_resoulution * radius }, y_hat { 0 };
+
+    /* The value of f(i)*f(i) */
+    int f2 { y * y };
+
+    /* 1st and 2nd order differences of f(i)*f(i) */
+    int delta_f2 { subpixel_resoulution * subpixel_resoulution },
+        delta2_f2 { -delta_f2 - delta_f2 };
```

Comment:
Another style nitpick, if the multiple assignments aren't fitting on a line, just break them into two:
```cpp
int delta_f2 = subpixel_resoulution * subpixel_resoulution;
int delta2_f2 = -delta_f2 - delta_f2;
```